### PR TITLE
Distribute label_replace

### PR DESCRIPTION
**Pull request by fpetkovski**
The label_replace function is not distributed because we don't handle string literals in the traversal function.

This commit fixes that.

### CODE REVIEW

**Comment by MichaHoffmann** [comment]
So I wonder what happens if you label replace an engine label away and execute a sum aggregation; is that problematic; can we add a test for it?

**Comment by fpetkovski** [comment]
This is a good catch. For now let's not distribute that case at all and emit a warning when a user runs this type of a query.

File: engine/engine.go [hunk 1]
```diff
@@ -330,7 +334,8 @@ func (q *compatibilityQuery) Exec(ctx context.Context) (ret *promql.Result) {
 		return &promql.Result{Value: promql.String{V: e.Val, T: q.ts.UnixMilli()}}
 	}
 	ret = &promql.Result{
-		Value: promql.Vector{},
+		Value:    promql.Vector{},
+		Warnings: q.warns,
```
Comment (MichaHoffmann) [inline engine/engine.go:338]: I think we need to merge here with the warnings from the context maybe?

Comment (fpetkovski) [reply engine/engine.go:336]: Done

**Review by MichaHoffmann — APPROVED** [review]
lgtm

**Pull request merged**